addons/base_import/models/base_import.py

```diff
@@ -1459,9 +1464,39 @@ def execute_import(self, fields, columns, options, dryrun=False):
         # convert load's internal nextrow to the imported file's
         if import_result['nextrow']: # don't update if nextrow = 0 (= no nextrow)
             import_result['nextrow'] += skip
+        if additional_filenames:
+            import_result['additional_filenames'] = additional_filenames
 
         return import_result
 
+    def _extract_additional_filenames(self, import_fields, data, model=False, prefix='', additional_filenames=False):
+        model = model or self.res_model
+        additional_filenames = additional_filenames or {}
+        all_fields = self.env[model].fields_get()
+        for name, field in all_fields.items():
+            name = prefix + name
+            if any(name + '/' in import_field and name == import_field.split('/')[prefix.count('/')] for import_field in import_fields):
+                # Recursive call with the relational as new model and add the field name to the prefix
+                data, additional_filenames = self._extract_additional_filenames(import_fields, data, field['relation'], name + '/', additional_filenames)
+            elif field['type'] == 'binary' and field.get('attachment') and any(f in name for f in IMAGE_FIELDS) and name in import_fields:
```

Comment:
`any(f in name for f in IMAGE_FIELDS)` is a crappy heuristic to determine if it is an Image field but I suppose it is preferable to mimic `_parse_import_data_recursive` anyway :+1: